### ISSUE TITLE
cpuspeed plugin does not work on Ubuntu 18.04.2 

### ISSUE THREAD

**Issue by Sanookmakmak** [munin-monitoring/munin#1173]
```
cd /etc/munin/plugins
ln -s /usr/share/munin/plugins/cpuspeed
munin-run cpuspeed
```

Error message:
```
awk: fatal: cannot open file `/sys/devices/system/cpu/cpu0/cpufreq/stats/time_in_state' for reading (No such file or directory)
awk: fatal: cannot open file `/sys/devices/system/cpu/cpu1/cpufreq/stats/time_in_state' for reading (No such file or directory)
```

`munin-node-configure --suggest|grep cpuspeed`

Error message:

`cpuspeed                   | yes  | no [missing /sys/devices/system/cpu/cpu0/cpufreq/stats/time_in_state]`

How can i fix it in Ubuntu 18.04.2 LTE?
Thank you!

**Comment by niclan** [comment]
The kernel support for this plugin appears to be somewhat on and off:

Intel i7 (2 years old) Ubuntu 18.10: No support
AMD Ryzen 5 (fall 2018) Ubuntu 18.10: Supported
Intel i5 (~10 years old) Ubuntu 16.04: Supported

I researched this 1-2 years ago and the results are a bit hazy to me. I _think_ it was dependent on the CPU scaling driver in use. The two machines I have that supports this plugin has this:

```
$ cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_driver
acpi-cpufreq
```
The result on the i7 which does not support the plugin is `intel_pstate`.

The good thing about the "time_in_state" file is that it provides a way to see the speed average over time without checking the file every few seconds. (In calculus terms it provides the integral of the time spent at each speed, so that the delta between each check of the file shows time spent at each speed step no matter how often or seldom you check the file, and therefore it provides a very good accounting of the real speed of the CPUs over time).

Making a plugin based on the contents of `/sys/devices/system/cpu/*/cpufreq/scaling_cur_freq` that samples the frequency every 5 minutes is quite useless because the frequency will always tend to be higher when the munin plugins run than at other times due to the higher load and therefore will not show anything like the true speed in any way - except if the machine always runs at full speed.

**Comment by sumpfralle** [comment]
Yes, it is a bit of a pity :(

Recently (released in v2.0.41) the plugin improved a bit:
* ace9a23aa14a314828b4b8a34b8db3bbea3168ee
* 75fd3f77319762c9ba74e01a443382469c70034e

The above commits introduce the usage of the (not really accurate) "snapshot" values of CPU frequency at the moment of the munin operation for CPUs lacking the fine-grained statistics.

Do you see anything more that could be improved here?

**Comment by sumpfralle** [comment]
Please reopen, if the problem is not fixed with 2.0.41 or later.
